core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -104,13 +105,14 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
   /**
    * Maps the error value of this effect to the specified constant value.
    */
-  final def asError[E1](e1: => E1): ZIO[R, E1, A] = mapError(new ZIO.ConstFn(() => e1))
+  final def asError[E1](e1: => E1)(implicit ev: CanFail[E]): ZIO[R, E1, A] =
+    mapError(new ZIO.ConstFn(() => e1))
 
   /**
    * Returns an effect whose failure and success channels have been mapped by
    * the specified pair of functions, `f` and `g`.
    */
-  final def bimap[E2, B](f: E => E2, g: A => B): ZIO[R, E2, B] = mapError(f).map(g)
+  final def bimap[E2, B](f: E => E2, g: A => B)(implicit ev: CanFail[E]): ZIO[R, E2, B] = mapError(f).map(g)
```

Comment:
This one makes sense but seems more trouble than it's worth, because the right hand side is always useful; so at worse this is "half useful".

Reply:
Hmm, I guess such `bimap` can always be replaced by `map`, so maybe it's useful to leave this one (sorry!).

Reply:
No problem! I struggled with the same thing on some of these.